Removing deprecated `isFullfilledBy()` method

nexxai opened the pull request:
<!-- Fill in the relevant information below to help triage your pull request. -->

|      Q       |   A
|------------- | -----------
| Type         | improvement
| Fixed issues | #5722 

#### Summary

Removing deprecated `isFullfilledBy()` method

nexxai committed:
Removing deprecated method

derrabus reviewed (changes requested):
Please update UPGRADE.md

nexxai committed:
Updated UPGRADE.md

nexxai commented:
@derrabus Sorry about that - I updated it for the 3.5.x deprecation but didn't realize it needed a separate line for the BC.  Have gone ahead and added that notice.

derrabus commented:
Thanks!
